File: src/main/java/com/manning/simplysend/mappers/OrderMapper.java
package com.manning.simplysend.mappers;

import java.util.stream.Collectors;
import java.util.stream.Stream;

import com.manning.simplysend.dto.ItemDTO;
import com.manning.simplysend.dto.OrderDTO;
import com.manning.simplysend.entities.Item;
import com.manning.simplysend.entities.Order;

public class OrderMapper extends BaseMapper {

    public static Order fromDTO(OrderDTO orderDTO) {
        Order order = new Order();
        order.setReason(orderDTO.getReason());
        order.setComment(orderDTO.getComment());
        order.setStatus(orderDTO.getStatus());

        Stream<ItemDTO> items = orderDTO.getItems().stream();
        order.setItems(items.map(ItemMapper::fromDTO).collect(Collectors.toList()));
        return order;
    }

    public static OrderDTO toDTO(Order order) {
        OrderDTO orderDTO = new OrderDTO();
        orderDTO.setId(order.getId());
        orderDTO.setReason(htmlEscape(order.getReason()));
        orderDTO.setComment(htmlEscape(order.getComment()));
        orderDTO.setStatus(order.getStatus());

        Stream<Item> items = order.getItems().stream();
        orderDTO.setItems(items.map(ItemMapper::toDTO).collect(Collectors.toList()));
        return orderDTO;
    }
}

File: src/main/kotlin/com/manning/simplysend/mappers/OrderMapper.kt
package com.manning.simplysend.mappers

import com.manning.simplysend.dto.OrderDTO
import com.manning.simplysend.entities.Order
import java.util.stream.Collectors

object OrderMapper : BaseMapper() {

    fun fromDTO(orderDTO: OrderDTO): Order {
        val order = Order()
        order.reason = orderDTO.reason
        order.comment = orderDTO.comment
        order.status = orderDTO.status

        val items = orderDTO.items!!.stream()
        order.items = items.map(ItemMapper::fromDTO).collect(Collectors.toList())

        return order
    }

    fun toDTO(order: Order): OrderDTO {
        val orderDTO = OrderDTO()
        orderDTO.id = order.id
        orderDTO.reason = htmlEscape(order.reason)
        orderDTO.comment = htmlEscape(order.comment)
        orderDTO.status = order.status

        val items = order.items!!.stream()
        orderDTO.items = items.map(ItemMapper::toDTO).collect(Collectors.toList())

        return orderDTO
    }
}